File: atlas_code_macos/macos/atlas_code_macos/Sources/atlas_code_macos/AtlasCodePlugin.swift
import FlutterMacOS
import Foundation

public class AtlasCodePlugin: NSObject, FlutterPlugin, AtlasCodeApi {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let binaryMessenger = registrar.messenger
    let instance = AtlasCodePlugin()
    AtlasCodeApiSetup.setUp(binaryMessenger: binaryMessenger, api: instance)
    registrar.publish(instance)
  }

  func localizedCountryNames(languageTag: String) throws -> [String: String] {
    let target = Locale(identifier: languageTag)
    var names = [String: String]()
    for code in NSLocale.isoCountryCodes {
      if let name = target.localizedString(forRegionCode: code) {
        names[code.uppercased()] = name
      }
    }
    return names
  }
}

File: atlas_code_android/android/src/main/kotlin/ni/devotion/atlas_code/AtlasCodePlugin.kt
package ni.devotion.atlas_code
import io.flutter.embedding.engine.plugins.FlutterPlugin
import java.util.Locale

class AtlasCodePlugin : FlutterPlugin, AtlasCodeApi {
    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        AtlasCodeApi.setUp(binding.binaryMessenger, this)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        AtlasCodeApi.setUp(binding.binaryMessenger, null)
    }

    override fun localizedCountryNames(languageTag: String): Map<String, String> {
        val target = Locale.forLanguageTag(languageTag)
        val names = mutableMapOf<String, String>()
        for (code in Locale.getISOCountries()) {
            val name = Locale.Builder().setRegion(code).build().getDisplayCountry(target)
            // getDisplayCountry echoes the code back when it has no
            // translation; omit those so Dart falls back to bundled names.
            if (name.isNotEmpty() && name != code) {
                names[code.uppercase(Locale.ROOT)] = name
            }
        }
        return names
    }
}
